### DISCUSSION TITLE
Play Or Retreat Based On % of winning.

### DISCUSSION THREAD

**Discussion by paperc07** [AdriaGual/marvel-snap-bot#64] [Ideas]
Maybe this is dumb, but is there anyway for the bot to be able to say "Ok, we've seen it discard an infinuat, hulk, etc and based on the cards that are out there they probably have hela" and retreat if it thinks it's beat? Or oh shit they discarded there hela im gonna snap? anything close to this type of logic possible? 

I mean we could even get a list of top decks from various sites idk if we would have to go to that extent.

**Comment by AdriaGual** [comment]
Difficult, you would need to:
- Detect enemy played cards
- Detect the exacte frame a card is discarded from the opponent and recognize which one has been discarded

With that then the logic would be simple but the detection part is difficult. But you can try, that's why I let the code open to everyone haha

**Reply by paperc07** [reply]
@AdriaGual I wish I knew how to code I was going to try and get CHAT GPT to help but it's down right now :(

**Reply by paperc07** [reply]
@AdriaGual  LMAO oh boy

<img width="642" alt="image" src="https://user-images.githubusercontent.com/9095222/217986098-7f1e3bed-06be-4046-927c-666fbd0487d6.png">

**Reply by paperc07** [reply]
this sounds so good but I just don't know how to input it lol 

<img width="601" alt="image" src="https://user-images.githubusercontent.com/9095222/217986420-ca70bf4e-5efc-4c88-bbe4-cf695fcd79f8.png">

**Reply by paperc07** [reply]
idk if any of this helps but who knows maybe we can figure it out using these examples

<img width="601" alt="image" src="https://user-images.githubusercontent.com/9095222/217988606-58fc7618-9b5d-4baa-9e0a-d441f72b6c48.png">

**Reply by AdriaGual** [reply]
ChatGPT looks seriously scary, seems like someone is gonna lose his job haha

Talking a bit more seriously, this modificacions are ok but if you take a look, it says that it won't save repeated cards so, in case you trigger swarm, and have 2 swarms in hand, this code will only save the first one.

Also it's not modifying the recognition algorithm, which is the main problem.

But props to ChatGPT, it explains the code really well!

**Reply by paperc07** [reply]
ahh ok well I tried, you think anything we can do from the other code i linked in other thread cause that was saying we could pull images from an image folder. wouldn't that help with the recognition or nah?
